usaspending_api/spending/v2/filters/spending_filter.py

```diff
@@ -0,0 +1,331 @@
+import logging
+import urllib
+from urllib.error import HTTPError
+
+from usaspending_api.awards.models import Award
+from usaspending_api.common.exceptions import InvalidParameterException
+
+logger = logging.getLogger(__name__)
+
+
+def spending_filter(alt_set, queryset, filters, _type):
+
+    for key, value in filters.items():
+        # check for valid key
+        if value is None:
+            raise InvalidParameterException('Invalid filter: ' + key + ' has null as its value.')
+
+        key_list = ['budget_function', 'budget_subfunction', 'federal_account', 'program_activity', 'object_class',
+                    'recipient', 'award', 'award_category', 'agency', 'agency_type', 'agency_sub', 'fy']
+
+        if key not in key_list:
+            raise InvalidParameterException(key + ' filter does not exist. '
+                                                  'Valid Filters: budget_function, budget_subfunction, federal_account,'
+                                                  'program_activity, object_class, recipient, award, award_category,'
+                                                  'agency, agency_type, agency_sub fy.')
+
+        # Check _type to filter on correct set (alt_set or queryset)
+        if _type == 'recipient' or _type == 'award' or _type == 'award_category' \
+                or _type == 'agency_type' or _type == 'agency_sub':
+            # Apply filters
+            # budget_function - DONE
+            if key == 'budget_function':
+                or_alt_set = None
+                if or_alt_set:
+                    or_alt_set |= or_alt_set.filter(
```

Comment:
Something @russellbodine and I discovered with our latest PR was that this method of or'ing querysets can lead to issues. Long story short, or'ing queries with itself doesn't merge the filters but doubles them and leads to issues. The solution we found that works is replacing the queryset variable with the `[original table].objects`. So for an example, `or_alt_set |= or_alt_set.filter(` to `or_alt_set |= FinancialAccountsByAwards.objects.filter(`.